DiagManager/CustomDiagnostics/SQL Server Perf Stats/SQL Server Perf Stats Snapshot.sql

```diff
@@ -164,107 +164,95 @@ begin
-			   (ar.secondary_role_allow_connections_desc is null or ars.role_desc='PRIMARY'))
-         order by sd.name
-	     
-	  END
-	  ELSE
-	  BEGIN
-	    
-		DECLARE dbCursor CURSOR Fast_Forward FOR 
-	    select name, database_id from sys.databases sd where sd.state_desc='ONLINE' and sd.database_id > 4 order by sd.name
-	  
-	  END
-	END
-	ELSE
-	BEGIN
-	
-	  DECLARE dbCursor CURSOR Fast_Forward FOR 
-	  select name, database_id from sys.databases sd where sd.state_desc='ONLINE' and sd.database_id > 4 order by sd.name
-	  
-	  	  
-	END
-
+	declare @dbname sysname, @dbid int	
+	DECLARE dbCursor CURSOR FOR 
+	select name, database_id from sys.databases where state_desc='ONLINE' and database_id > 4 order by name
 	OPEN dbCursor
 
 	FETCH NEXT FROM dbCursor  INTO @dbname, @dbid
-
-	select @dbid 'Database_Id', @dbname 'Database_Name',  Object_name(st.object_id) 'Object_Name',  st.* into #tmpStats 
-	from sys.stats ss cross apply sys.dm_db_stats_properties (ss.object_id, stats_id) st --replaced sys.dm_db_index_usage_stats  by sys.stat since the first doesn't return anything in case the table or index was not accessed since last SQL restart
-	where 1=0 
-	
-
+	--replaced sys.dm_db_index_usage_stats  by sys.stat since the first doesn't return anything in case the table or index was not accessed since last SQL restart
+    select @dbid 'Database_Id', @dbname 'Database_Name',  Object_name(st.object_id) 'Object_Name', SCHEMA_NAME(schema_id) 'Schema_Name', st.* into #tmpStats from sys.stats ss cross apply sys.dm_db_stats_properties (ss.object_id, ss.stats_id) st inner join sys.objects so ON (ss.object_id = so.object_id) where 1=0
 	WHILE @@FETCH_STATUS = 0
 	begin
 	
-		declare @sql nvarchar (512)
+		declare @sql nvarchar (max)
 		set @sql = 'USE [' + @dbname + ']'
-	
-		set @sql = @sql + '	insert into #tmpStats	
-		                    select ' + cast( @dbid as nvarchar(20)) +   ' ''Database_Id''' + ',''' +  @dbname  + ''' Database_Name,  Object_name(st.object_id) ''Object_Name'',  st.* 
-		                    from sys.stats ss  
-							  cross apply sys.dm_db_stats_properties (ss.object_id, ss.stats_id) st 
-		                      inner join sys.objects so ON (ss.object_id = so.object_id) where so.type_desc not in (''SYSTEM_TABLE'', ''INTERNAL_TABLE'')'  
-							  --replaced sys.dm_db_index_usage_stats  by sys.stat since the first doesn't return anything in case the table or index was not accessed since last SQL restart
-							  
-		exec (@sql)
+	    --replaced sys.dm_db_index_usage_stats  by sys.stat since the first doesn't return anything in case the table or index was not accessed since last SQL restart
+		set @sql = @sql + '	insert into #tmpStats	select ' + cast( @dbid as nvarchar(20)) +   ' ''Database_Id''' + ',''' +  @dbname  + ''' Database_Name,  Object_name(st.object_id) ''Object_Name'', SCHEMA_NAME(schema_id) ''Schema_Name'', st.* from sys.stats ss 
```

Comment:
Can you consider adding the columns explicitly and replace * in this query 
  ` st.* from sys.stats ss `
That way you control what SQL Nexus imports. Also check for any other places in this stats query that you replaced.

Reply:
Thanks for the heads up! The modification has been performed